libs/dal/src/repositories/notification/notification.schema.ts

```diff
@@ -34,9 +34,6 @@ const notificationSchema = new Schema<NotificationDBModel>(
         type: Schema.Types.String,
       },
     ],
-    _digestedNotificationId: {
-      type: Schema.Types.String,
-    },
```

Comment:
@p-fernandez  Thank you for the detailed feedback, I will reply one by one... coming to DAG, it is Directed acyclic graph, from sequence of steps creating a parallel execution graph, example, all digest steps should start parallel, it does not make sense to run those sequentially, delay steps with delaypath also should run parallel, sequence does not make sense for those jobs as well, because we do not know what users pass for the delaypath.  Coming to trigger jobs, those are unnecessary in this new design.. and delay jobs alos I felt unnecessary, delay should be an attribute of the job/event, look at any queue system/calendar, the delay is always associated with the job/event, that is why I felt it is unnecessary and propagated delay to next active step. I believe for end users also having delay part of jobs is more comfortable and intuitive(probably take a poll or feedback?), but in any case if we keep delay on the front end also the api code works fine, with out breaking anything we just skipping delay jobs behind the scenes. Also can you let me know which cases it is not working?  did you try this on latest code that has digestRunKey? I tried all possible ways and it is working fine for me. If you can share the script I can try it and check which cases failing.. I also uncommented some of the old test cases commented in old code and working fine in new code and also created parallel dynamic generation of test cases by running parallel execution of 300 events with multiple subscribers. if you can share the script then I can try and fix it if any issues there.. As it is draft PR, I was just focusing on the API changes, if this direction is fine, I can create final PR and that time I can have complete impact analysis and fix other parts of the code as well, also looks like lot of changes happening on the next branch side, I have adjusted some code, but couple of cases skipped it(will discuss more on this later), yes backward compatibility I totally agree, I believe all these changes we can do with out breaking old functionality, even details logs also mostly taken care, if any details logs missing we can add easily in final PR.
Here are the specs followed in this PR:
1. Digest and delay with path always at the root level of DAG
2. Delays are added to next active step and delay steps removed
3. Two consecutive delays adds up and added to next active step
4. Delays just before digest are ignored
5. Delays at the leaf level of branches ignored
6. Steps before digest forms a parallel branch
7. Delay path step forms parallel branch
8. two consecutive digest steps forms two branches, one branch has just digest node
9. Negative delays in meta should throw error
10. Negative/missing delays in payload and overrides should ignore and proceed with delay 0
11. Missing delay path in meta should throw error for delay nodes
12. If any inactive step present in the branch then all the steps below will be trimmed

Some of these specs based on current code/docs and some places where we don't have clear doc in existing system, made up my own, If we need to change any of these please let me know, we can adjust code very easily

Reply:
Sure In final PR we can adjust this code easily, instead of _digestedNotificationId we just need to use the _digestedNotificationIds array and we can remove _digestedNotificationId  column., small change.